lib/resourcemerge/core_test.go

```diff
@@ -30,11 +30,12 @@ func TestEnsurePodSpec(t *testing.T) {
 			expected:         corev1.PodSpec{},
 		},
 		{
-			name: "remove regular containers from existing",
+			name: "remove regular containers from existing, no securityContext",
```

Comment:
the clearing-on-unset logic is being exercised in `container securityContext none`, right?  We may be able to leave this case focused on removing containers.

Reply:
Correct. The ones w/o `Container` in title are for `PodSpec`. I clarify names.

Reply:
added